test/videojs-hls_test.js

```diff
@@ -1724,6 +1724,41 @@ test('calling play() at the end of a video resets the media index', function() {
   strictEqual(player.hls.mediaIndex, 0, 'index is 1 after the first segment');
 });
 
+test('drainBuffer will not proceed with empty source buffer', function() {
+  var oldMedia, newMedia, compareBuffer;
+  player.src({
+    src: 'https://example.com/encrypted-media.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  openMediaSource(player);
+
+  oldMedia = player.hls.playlists.media;
+  newMedia = {segments: [{
+    key: {
```

Comment:
Making the segment a key makes the bug-infected code throw an error sooner. It gets to the segmentBuffer.shift() without having to override other function calls that happen in drainBuffer() and are irrelevant to the test, and I think it's the lesser of two evils.